src/libcore/macros.rs

```diff
@@ -182,7 +182,9 @@ macro_rules! try {
 /// ```
 #[macro_export]
 macro_rules! write {
-    ($dst:expr, $($arg:tt)*) => ($dst.write_fmt(format_args!($($arg)*)))
+    ($dst:expr, $($arg:tt)*) => (
+        ::std::io::Write::write_fmt($dst, format_args!($($arg)*))
```

Comment:
This should use `$crate::io::Write::write_fmt` instead of `::std::....` to avoid code trying to use this directly from libcore.